src/main/front/src/pages/CoursePage.tsx

```
@@ -42,10 +130,30 @@ function CoursePage() {
+                const calculatedProgress: UserProgress[] =
+                  calculateProgress(programProcess);
+                const myProgress = calculatedProgress.find(
+                  (u) => u.userId === userId
+                );
+                const courseId = courseData.id;
+                const courseProgress = myProgress?.courseProgress[courseId];
+                if (courseProgress && courseProgress.map) {
+                  const state = courseProgress.map[group.id];
+                  isCompleted = state === "DONE";
+                  isInProgress = state === "IN_PROGRESS";
+                }
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)